google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferManager.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.transfermanager;
+
+import com.google.cloud.storage.BlobInfo;
+import com.google.common.base.MoreObjects;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+public interface TransferManager {
+
+  class TransferManagerConfig {
+    @NonNull private final int maxWorkers;
+    @NonNull private final int perWorkerBufferSize;
```

Comment:
Thoughts on changing this to `maxTotalBufferSize` which could leave us wiggle room on how much per worker we use if we're able to determine scaling up buffer size can increase throughput.

Reply:
I feel like we are moving further from the original design which was chunkSize if we continue in this direction. I don't think this is necessarily a bad thing but node and python are also out already with chunkSize.